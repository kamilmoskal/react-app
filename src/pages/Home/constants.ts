import { OnTheAirState } from 'src/store/TVShow/OnTheAir/constants';

interface HomeOwnProps { };

type HomeStateProps = {
    onTheAir: OnTheAirState
}

export type HomeProps = HomeOwnProps & HomeStateProps