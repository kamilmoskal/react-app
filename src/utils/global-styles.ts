import { createGlobalStyle } from "styled-components";

const GlobalStyle = createGlobalStyle`
    * {
        box-sizing: border-box;
        transition: all 0.2s linear;
        transition-property: color, background-color, border-color;
    }

    body {
        background: yellow;
    }

    a {
        color: inherit;
        text-decoration: none;
    }

`;

export default GlobalStyle;