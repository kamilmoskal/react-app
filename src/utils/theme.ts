import baseStyled, { ThemedStyledInterface } from 'styled-components';

export const lightTheme = {
    all: {
        borderRadius: '0.5rem',
    },
    main: {
        color: '#FAFAFA',
        textColor: '#212121',
        bodyColor: '#FFF',
    },
    secondary: {
        color: 'blue',
    },
};

export const darkTheme: Theme = {
    all: { ...lightTheme.all },
    main: {
        color: '#212121',
        textColor: '#FAFAFA',
        bodyColor: '#424242',
    },
    secondary: {
        color: 'red',
    },
};

export type Theme = typeof lightTheme;
export const styled = baseStyled as ThemedStyledInterface<Theme>;