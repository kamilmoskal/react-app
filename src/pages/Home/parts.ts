import { styled } from 'src/utils/theme';

export const Wrapper = styled.div`
    ${({ theme: { secondary } }) => `color: ${secondary.color}`}
`