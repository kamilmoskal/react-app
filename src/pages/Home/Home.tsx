import React from 'react';
import { connect } from 'react-redux';
import Link from 'next/link'
import Head from 'next/head';
import { Store } from 'src/store/rootReducer';
import * as P from './parts';
import { HomeProps } from './constants';

export const Home: React.FC<HomeProps> = ({ onTheAir: { data, isLoading } }) => (
    <>
        <Head>
            <title key="title">Strona główna</title>
            <meta name="description" content="Strona główna bez prawie żadnej zawartości" key="description" />
            <meta name="keywords" content="strona, główna, brak zawartości" key="keywords" />
        </Head>
        <P.Wrapper>
            <h1>Hello world! - user agent: asd</h1>
            <p><Link href={'/auth'}><a>link to auth</a></Link></p>
            <h1>store:{isLoading}</h1>
            {data && data.results && data.results.map((result, i) => <p key={i}>{result.original_name}</p>)}
        </P.Wrapper>
    </>
)

const mapStateToProps = (store: Store) => ({
    onTheAir: store.onTheAir
})

export default connect(mapStateToProps)(Home);