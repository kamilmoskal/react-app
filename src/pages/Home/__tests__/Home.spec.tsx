import "jest-styled-components";
import React from "react";
import { ThemeProvider } from "styled-components";
import renderer from 'react-test-renderer';
import { shallow } from "enzyme";
import { lightTheme } from 'src/utils/theme';
import { Home } from '../Home';

/* random tests to check if it works */
describe("Home", () => {
  it('should render correctly', () => {
    const component = renderer.create(
      <ThemeProvider theme={lightTheme}>
        <Home onTheAir={{ isLoading: false }} />
      </ThemeProvider>
    ).toJSON()
    expect(component).toMatchSnapshot()
  })

  it('should shows "link to auth" in a <p> tag', () => {
    const app = shallow(<Home onTheAir={{ isLoading: false }} />);
    expect(app.find("a").text()).toEqual("link to auth");
  });
});