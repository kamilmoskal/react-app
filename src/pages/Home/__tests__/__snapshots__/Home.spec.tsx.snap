// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
.c0 {
  color: blue;
}

<div
  className="c0"
>
  <h1>
    Hello world! - user agent: asd
  </h1>
  <p>
    <a
      href="/auth"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      link to auth
    </a>
  </p>
  <h1>
    store:
  </h1>
</div>
`;
